docs/src/references/tedge-log-management.md

```diff
@@ -0,0 +1,232 @@
+* Each entry in the the `tedge-log-plugin.toml` file contains a log `type` and a `path` pattern,
+  where the `type` is used to represent the logical group of log files matching the `path` pattern.
+* On receipt of a log file request for a given `type`, 
+  the log files for that type are retrieved using the `path` pattern defined in this `tedge-log-plugin.toml`,
+  matched against the requested time range, search text and maximum line count.
+* The list of managed log files in `tedge-log-plugin.toml` can be updated both locally as well as from cloud, for instance,
+  using the configuration management feature of Cumulocity, combined with the `tedge-configuration-plugin` of thin-edge.
+
+## Installation
+
+As part of this plugin installation:
+* On systemd enabled devices, the service definition file for this `tedge-log-plugin` daemon is also installed as part of this plugin installation.
+
+Once installed, the `tedge-log-plugin` is run as a daemon on the device listening to log requests from cloud mapper on `te/device/main///cmd/log_upload/+` MQTT topic.
+
+## Configuration
+
+The `tedge-log-plugin` configuration is stored by default under `/etc/tedge/plugins/tedge-log-plugin.toml`.
+
+This [TOML](https://toml.io/en/) file defines the list of log files that can be retrieved from the cloud tenant.
+The paths to these files can be represented using [glob](https://en.wikipedia.org/wiki/Glob_(programming)) patterns.
+The `type` given to these paths are used as the log type associated to a log path.
+
+```toml title="file: /etc/tedge/plugins/tedge-log-plugin.toml"
+files = [
+  { type = "mosquitto", path = '/var/log/mosquitto/mosquitto.log' },
+  { type = "software-management", path = '/var/log/tedge/agent/software-*' },
+  { type = "c8y_CustomOperation", path = '/var/log/tedge/agent/c8y_CustomOperation/*' }
+]
+```
+
+The `tedge-log-plugin` parses this configuration file on startup for all the `type` values specified,
+and sends the supported log types message to MQTT local broker on `te/device/main///cmd/log_upload` topic as follows:
+
+```json
+{
+  "types" : [ "mosquitto", "software-management", "c8y_CustomOperation" ]
+}
+```
+
+The plugin continuously watches this configuration file for any changes and resends the JSON message with the `type`s in this file,
+whenever it is updated.
+
+:::note
+If the file `/etc/tedge/plugins/tedge-log-plugin.toml` is ill-formed or cannot be read,
+then a JSON message with an empty array for the `types` field is sent, indicating no log files are tracked.
+:::
+
+## Startup
+
+The plugin declares **supported operation** and **supported log types** on startup.
+It's up to cloud mappers, if it converts the declaration messages to the corresponding cloud format or discard them.
+
+### Supported operation
+
+On startup, the plugin declares to the Cumulocity mapper that it supports `c8y_LogfileRequest` operation
+by creating an empty file at `/etc/tedge/operations/c8y/c8y_LogfileRequest`.
```

Comment:
The mapper should be creating that file on receipt of the tedge supported log types message described above and below.